### PR TITLE
Add AE TypeDefs for AppKit Apps

### PR DESCRIPTION
**Pull request by CKegel**
adds required typedefs in AppleEvents so that TextEdit will compile and run

### CODE REVIEW

File: src/frameworks/CoreServices/include/AE/AE.h [hunk 1]
```diff
@@ -6,4 +6,20 @@
...
+kAEQuitApplication = 'quit',
+kAEAnswer = 'ansr',
+kAEApplicationDied = 'obit',
+kAEShowPreferences = 'pref',
+};
+
+enum : DescType {
+    kCoreEventClass = 'aevt',
+};
+
```
Comment (CuriousTommy) [inline src/frameworks/CoreServices/include/AE/AE.h:24]: These declarations should be in `AppleEvents.h` instead. That's where they live in Apple's SDK.

**Review by CuriousTommy — CHANGES_REQUESTED** [review]

File: src/frameworks/CoreServices/include/AE/AE.h (hunk 1 above)
Comment (CKegel) [reply src/frameworks/CoreServices/include/AE/AE.h:24]: Does `AppleEvents.h` currently exist, or should I create it in `src/frameworks/include/AE/`?

Comment (CuriousTommy) [reply src/frameworks/CoreServices/include/AE/AE.h:24]: You can create one at `src/frameworks/include/AE`.

Comment (CKegel) [reply src/frameworks/CoreServices/include/AE/AE.h]: I think I've done it correctly - let me know if anything else needs changed!

**Comment by CuriousTommy** [comment]
Thank you for you contribution!

**Pull request merged**